File: src/main/java/io/github/thehrz/snowcraft/object/misc/SnowGolemSuit.java
package io.github.thehrz.snowcraft.object.misc;

import io.github.thehrz.snowcraft.list.Items;
import io.izzel.taboolib.module.inject.TListener;
import me.mrCookieSlime.Slimefun.Setup.SlimefunManager;
import org.bukkit.Material;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerMoveEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;

@TListener
public class SnowGolemSuit implements Listener {
    @EventHandler
    public void onPlayerMove(PlayerMoveEvent event) {
        if (event.getPlayer().getInventory().getHelmet() != null &&
                event.getPlayer().getInventory().getChestplate() != null &&
                event.getPlayer().getInventory().getLeggings() != null &&
                event.getPlayer().getInventory().getBoots() != null &&
                SlimefunManager.isItemSimiliar(event.getPlayer().getInventory().getHelmet(), Items.SNOW_GOLEM_HELMET, true) &&
                SlimefunManager.isItemSimiliar(event.getPlayer().getInventory().getChestplate(), Items.SNOW_GOLEM_CHESTPLATE, true) &&
                SlimefunManager.isItemSimiliar(event.getPlayer().getInventory().getLeggings(), Items.SNOW_GOLEM_LEGGINGS, true) &&
                SlimefunManager.isItemSimiliar(event.getPlayer().getInventory().getBoots(), Items.SNOW_GOLEM_BOOTS, true) &&
                event.getPlayer().getLocation().getBlock().getType() == Material.SNOW ||
                event.getPlayer().getLocation().getBlock().getType() == Material.SNOW_BLOCK) {
            event.getPlayer().addPotionEffect(new PotionEffect(PotionEffectType.SPEED, 200, 2));
        }
    }
}

File: src/main/kotlin/io/github/thehrz/snowcraft/object/misc/SnowGolemSuit.kt
package io.github.thehrz.snowcraft.`object`.misc

import io.github.thehrz.snowcraft.list.Items
import me.mrCookieSlime.Slimefun.Setup.SlimefunManager
import org.bukkit.Material
import org.bukkit.event.player.PlayerMoveEvent
import org.bukkit.potion.PotionEffect
import org.bukkit.potion.PotionEffectType
import taboolib.common.platform.event.SubscribeEvent

object SnowGolemSuit {
    @SubscribeEvent
    fun e(e: PlayerMoveEvent) {
        if (e.player.inventory.helmet != null && e.player.inventory.chestplate != null && e.player.inventory.leggings != null && e.player.inventory.boots != null &&
            SlimefunManager.isItemSimiliar(e.player.inventory.helmet, Items.SNOW_GOLEM_HELMET, true) &&
            SlimefunManager.isItemSimiliar(e.player.inventory.chestplate, Items.SNOW_GOLEM_CHESTPLATE, true) &&
            SlimefunManager.isItemSimiliar(e.player.inventory.leggings, Items.SNOW_GOLEM_LEGGINGS, true) &&
            SlimefunManager.isItemSimiliar(e.player.inventory.boots, Items.SNOW_GOLEM_BOOTS, true) &&
            (e.player.location.block.type == Material.SNOW || e.player.location.block.type == Material.SNOW_BLOCK)
        ) {
            e.player.addPotionEffect(PotionEffect(PotionEffectType.SPEED, 200, 2))
        }
    }
}